### ISSUE TITLE
Assisted inject in jdk8+ compiled with -parameters not require @Assisted with parameter

### ISSUE THREAD

**Issue by brunoais** [google/guice#1142]
Is there a way or is there a feature in guice that allows, when jdk8 is used and `-parameters` flag is used with the compiler, guice can correlate the builder's methods with the right constructor based only on the names of the parameters which get stored in the `.class` file?

**Comment by dimo414** [comment]
I imagine this does not exist. Personally relying on a `javac` flag for correct behavior here would be a poor experience (you'd get runtime failures based on whether or not you remember the `-parameters` flag) but you could make a case for it being added. Could you rephrase your initial post as a feature request rather than a question, and include more details about the behavior you'd expect - notably how this should handle `-parameters` *not* being specified.

**Comment by brunoais** [comment]
@dimo414 That's my initial proposal with the limited knowledge I have of Guice.
Please comment on it and call for attention to problems I didn't notice.

**Comment by brunoais** [comment]
@dimo414 Do you know if my suggestion can be applied? If so, any predictions?
Or
May I do it?

**Comment by brunoais** [comment]
@dimo414 I didn't get any response in almost 1 month. What should I do?

**Comment by brunoais** [comment]
@ronshapiro Any comments about this?
